document_collection.rb

```diff
@@ -15,4 +15,8 @@ def add_document(document)
 	def size
 		@arr.size
 	end
+
+	def element(i)
```

Comment:
Maybe switch this to something like

```
def [](key)
  @arr[key]
end
```

You could then access the document_collection variable with the normal bracket notation. Ie

Instead of `dc.element(my_thing)` you could use `dc[my_thing]`.